frontend/src/components/UserMenu.jsx

```diff
@@ -30,17 +31,37 @@ class UserMenu extends React.Component {
   };
 
   handleLogout = () => {
-    this.setState({
-      auth: false,
-      anchorEl: null
-    });
+    fetch('http://localhost:6543/login', {
+      method: 'DELETE',
+      headers: {
+        'x-auth-token': localStorage.getItem('token')
+      }
+    }).then(res => {
+        localStorage.removeItem('token');
```

Comment:
You'd better check response from backed whether it was successful or not before cleaning auth info on frontend side.